src/mrack/transformers/openstack.py

```diff
@@ -144,15 +144,23 @@ def _get_network_type(self, host):
         return network_type
 
     def _get_image(self, os):
-        """Get image name by OS name from provisioning config."""
-        return get_config_value(self.config["images"], os)
+        """
+        Get image name by OS name from provisioning config.
+
+        Returns:
+            1. image by the os key
+            2. default from the images if os is not found in keys
+            3. os name if default is not specified for images.
+        """
+        return get_config_value(self.config["images"], os, os)
 
     def create_host_requirement(self, host):
         """Create single input for OpenStack provisioner."""
+        required_image = host.get("image") or self._get_image(host["os"])
```

Comment:
Does it conflict with `validate_hosts()`?

Reply:
Nice catch! When the image would be specified in metadata and will not be present in provisionin-config that would be a problem.